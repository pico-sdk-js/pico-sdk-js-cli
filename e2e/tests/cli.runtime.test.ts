import {describe, it} from '@jest/globals';

import psjRunner from './psjRunner';

describe('PSJ Runtime Scenarios', () => {

    describe('repl', () => {

        it('executes javascript', async () => {

            await psjRunner()
            .start(['--auto-connect', '--skip-header'])
            .command(`const y = 1;`)
            .command(`y`)
            .assertSnapshot();
        });
    });

    describe('.restart', () => {

        it('soft will resets all data', async () => {

            await psjRunner()
            .start(['--auto-connect', '--skip-header'])
            .command(`const y = 1;`)
            .command(`.restart`)
            .command(`y`)
            .assertSnapshot();
        });

        it('hard will disconnect & reset data', async () => {

            await psjRunner()
            .start(['--auto-connect', '--skip-header'])
            .command(`const y = 1;`)
            .command(`.restart --hard`)
            .pause(1000)
            .command(`.connect`)
            .command(`y`)
            .assertSnapshot();
        });
    });
});