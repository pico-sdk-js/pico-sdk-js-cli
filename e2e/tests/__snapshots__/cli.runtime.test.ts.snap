// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PSJ Runtime Scenarios .restart hard will disconnect & reset data: stdOut 1`] = `
"Connecting ... 
Connecting to serial device at /dev/ttyACM0
> const y = 1;
> .restart --hard
Device restarting
Disconnecting ... 
> .connect
Connecting ... 
Connecting to serial device at /dev/ttyACM0
> y
ERR: y is not defined
> "
`;

exports[`PSJ Runtime Scenarios .restart soft will resets all data: stdOut 1`] = `
"Connecting ... 
Connecting to serial device at /dev/ttyACM0
> const y = 1;
> .restart
Device restarting
> y
ERR: y is not defined
> "
`;

exports[`PSJ Runtime Scenarios repl executes javascript: stdOut 1`] = `
"Connecting ... 
Connecting to serial device at /dev/ttyACM0
> const y = 1;
> y
1
> "
`;
